integration_test/flash_tested/Flash.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef HAL_ST_INTEGRATION_TEST_FLASH_HPP
+#define HAL_ST_INTEGRATION_TEST_FLASH_HPP
+
+#include "generated/echo/Flash.pb.hpp"
+#include "hal/interfaces/FlashHomogeneous.hpp"
+#include "infra/util/AutoResetFunction.hpp"
+
+namespace application
+{
+    class Flash
+        : public flash::Flash
+    {
+    public:
+        Flash(services::Echo& echo, hal::Flash& flash);
+
+        // Implementation of flash::Flash
+        void Read(uint32_t address, uint32_t size) override;
+        void Write(uint32_t address, infra::ConstByteRange contents) override;
+        void EraseSectors(uint32_t sector, uint32_t numberOfSectors) override;
+
+    private:
+        flash::FlashResultProxy flashResult;
+        hal::Flash& flash;
+
+        std::array<uint8_t, std::tuple_size<std::remove_reference_t<decltype(std::declval<flash::WriteRequest>().contents.Storage())>>::value> buffer;
```

Comment:
Could you extract 'std::tuple_size<std::remove_reference_t<decltype(std::declval<flash::WriteRequest>().contents.Storage())>>::value' for readability, as it is used multiple times.

Reply:
I've now generated "contentsSize" by the protobuf plugin